src/pages/Login/Login.jsx

```diff
@@ -1,5 +1,211 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
 
 export default function Login() {
-  return <div></div>;
+  const [emailValue, setEmailValue] = useState('');
+  const [passwordValue, setPasswordValue] = useState('');
+  const [isEmailValid, setIsEmailValid] = useState('');
+  const [isPasswordValid, setIsPasswordValid] = useState('');
+
+  const controlValue = event => {
+    if (event.target.name === 'inputEmail') {
+      setEmailValue(event.target.value);
+    } else if (event.target.name === 'inputPassword') {
+      setPasswordValue(event.target.value);
+    }
+  };
+
+  const allValueCheck = () => {
+    const reg_email = /^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/;
+
+    const reg_pwd =
+      /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%#?&])[A-Za-z\d@$!%*#?&]{8,16}$/;
+
+    setIsEmailValid(reg_email.test(emailValue));
+    console.log('이메일 정규식 확인', isEmailValid);
+
+    setIsPasswordValid(reg_pwd.test(passwordValue));
+
+    console.log('비밀번호 정규식 확인', isPasswordValid);
+  };
+
+  const goToMain = e => {
+    e.preventDefault();
+
+    if (!isEmailValid && !isPasswordValid) {
+      return;
+    }
+
+    fetch(`http://ambitiouskyle.iptime.org:6389/users/signin`, {
+      method: 'post',
+      body: JSON.stringify({
+        email: emailValue,
+        password: passwordValue,
+      }),
+    })
+      .then(response => response.json())
+      .then(result => {
+        //임의로 작성
+        if (result.message === 'SUCCESS') {
+          localStorage.setItem('TOKEN', result.access_token);
+        } else {
+          alert('잘못된 정보입니다. 다시 입력해주세요!');
+        }
+      });
+  };
+
+  // 두 값 모두 유효할떄 버튼 디스에이블 = false 로  만들기
+  //
+  const isAllValueValid = () => {
+    const isAllValueValid = isEmailValid && isPasswordValid;
+    return isAllValueValid;
+  };
```

Comment:
```suggestion
  const isAllValueValid = isEmailValid && isPasswordValid;
```
- 함수 보다는 변수에 바로 boolean 데이터 값을 담는 것도 좋을 것 같아요. 아래 코드 리뷰 내용과 같이 봐주세요.

Reply:
생각해보니  Login() 자체가 함수여서 내부에 바로 const isAllValueValid = isEmailValid && isPasswordValid; 로 선언을 해서 쓸 수 있는거였네요 ! 리뷰 감사합니다